src/tokens/Assets.tsx

```diff
@@ -268,18 +286,91 @@ function AssetsScreen({ navigation, route }: Props) {
     return null
   }
 
-  const keyExtractor = (item: TokenBalance | Position) => {
+  const keyExtractor = (item: TokenBalance | Position | Nft) => {
     if (assetIsPosition(item)) {
       return item.address
+    } else if ('balance' in item) {
+      return item.tokenId
+    } else {
+      return `${item.contractAddress}-${item.tokenId}`
     }
-    return item.tokenId
   }
 
-  const renderAssetItem = ({ item }: { item: TokenBalance | Position }) => {
+  const NftItem = ({ item, index }: { item: Nft; index: number }) => {
+    return (
+      <View
+        testID="NftItem"
+        style={[
+          styles.nftsTouchableContainer,
+          // For even indexes, add right margin; for odd indexes, add left margin.
+          // If the index is even and it's the last image, add a right margin to left-align the image in the last row.
+          index % 2 === 0
+            ? { marginRight: Spacing.Regular16 } &&
+              index === nfts.length - 1 &&
+              styles.nftsLastImage
```

Comment:
Not sure I understand this. Is the right margin added for even indices if its not the last item? It seems like `index === nfts.length - 1` would be false and this would ultimately result in a `false`. The right margin seems to be applied only for the last item (from `styles.nftsLastImage`) which also seems like it shouldn't  make any difference since the last row will have only one item in this case.

Reply:
If the last image is an even index we only need to apply the right margin. If the last image is an odd index we need to apply special style to align it left, otherwise we get cases where the last image is in the center of the two columns.